### ISSUE TITLE
Prevent ExtraField from overriding existing data

### ISSUE THREAD

**Issue by grundic** [prawn-cake/simple-models#24]
Here is an example:

```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-

from simplemodels.fields import CharField
from simplemodels.models import Document


class User(Document):
    class Meta:
        ALLOW_EXTRA_FIELDS = True

    name = CharField()

    def id(self):
        return 'some-name'


if __name__ == '__main__':
    user = User(dict(name='foobar', id=888))
    assert callable(user.id)

    >>> AssertionError
```

Right now, `ExtraField` is created unconditionally, what may cause overwriting existing data (functions, attributes, whatever). Need a way to prevent this situation.

**Comment by prawn-cake** [comment]
It would be reasonable to throw an exception on document creation step if it found an overlapping property.
I'll take a look as well. I guess there are another corner cases, e.g. if a field and a property have the same name

**Comment by prawn-cake** [comment]
Fixed in [v0.6.1](https://github.com/prawn-cake/simple-models/tree/0.6.1)
